### DISCUSSION TITLE
Version number scheme

### DISCUSSION THREAD

**Discussion by jbeyerstedt** [mavlink-router/mavlink-router#344] [Ideas]
In the past, mavlink-router had just two tagged releases, which got old quite soon. This lead to some confusion when people used the release instead of the latest master branch and had issues, which already got fixed in the master.

We're currently in the final steps to release version 3. A new major version is justified because many things changed internally and feature-wise, so a minor version wouldn't reflect this properly.

But I wanted to start a discussion here, if we should:
a) just continue increasing the "major" version number (hopefully more often than once every few years) or
b) switch to semantic versioning and do minor releases with minor feature additions as well as patch-releases when some number of bugfixes were merged (every few months)

**Comment by lucasdemarchi** [comment]
(a). I don't like the major.minor - we just don't break user compatibility unless it's broken beyond repair or it's very niche thing that nobody uses.

Yes, we should make releases much more often. And we can have a stable branch from which we release version.<patch>, but I'm not sure we have enough audience to deserve that - we could as well just release a new version when needed.

**Reply by jbeyerstedt** [reply]
OK, just wanted to be clear about the numbering scheme.

I don't think, that we need a separate stable and dev branch since the master branch is pretty stable from what I can tell. It might make sense if we had a lot more testing going on, which couldn't be integrated into the PR's CI runs. But we don't have the resources (financially and especially developer time) for that.
